src/core/server/http/integration_tests/http_auth.test.ts

```diff
@@ -0,0 +1,249 @@
+        const { http } = await root.setup();
+        const { registerAuth, createRouter, auth } = http;
+
+        registerAuth((req, res, toolkit) => toolkit.authenticated());
+
+        const router = createRouter('');
+        registerRoute(router, auth, false);
+
+        await root.start();
+        await kbnTestServer.request.get(root, '/route').expect(200, { authenticated: false });
+      });
+
+      it('allows anonymous access when auth returns `notHandled`', async () => {
+        const { http } = await root.setup();
+        const { registerAuth, createRouter, auth } = http;
+
+        registerAuth((req, res, toolkit) => toolkit.notHandled());
+
+        const router = createRouter('');
+        registerRoute(router, auth, false);
+
+        await root.start();
+        await kbnTestServer.request.get(root, '/route').expect(200, { authenticated: false });
+      });
+
+      it('allows anonymous access when auth returns `unauthorized`', async () => {
+        const { http } = await root.setup();
+        const { registerAuth, createRouter, auth } = http;
+
+        registerAuth((req, res, toolkit) => res.unauthorized());
+
+        const router = createRouter('');
+        registerRoute(router, auth, false);
+
+        await root.start();
+        await kbnTestServer.request.get(root, '/route').expect(200, { authenticated: false });
+      });
+    });
+    describe('when authRequired is `optional`', () => {
+      it('allows authenticated access when auth returns `authenticated`', async () => {
+        const { http } = await root.setup();
+        const { registerAuth, createRouter, auth } = http;
+
+        registerAuth((req, res, toolkit) => toolkit.authenticated());
+
+        const router = createRouter('');
+        registerRoute(router, auth, 'optional');
+
+        await root.start();
+        await kbnTestServer.request.get(root, '/route').expect(200, { authenticated: true });
+      });
+
+      it('allows anonymous access when auth returns `notHandled`', async () => {
+        const { http } = await root.setup();
+        const { registerAuth, createRouter, auth } = http;
+
+        registerAuth((req, res, toolkit) => toolkit.notHandled());
+
+        const router = createRouter('');
+        registerRoute(router, auth, 'optional');
+
+        await root.start();
+        await kbnTestServer.request.get(root, '/route').expect(200, { authenticated: false });
+      });
+
+      it('blocks access when auth returns `unauthorized`', async () => {
+        const { http } = await root.setup();
+        const { registerAuth, createRouter, auth } = http;
+
+        registerAuth((req, res, toolkit) => res.unauthorized());
+
+        const router = createRouter('');
+        registerRoute(router, auth, 'optional');
+
+        await root.start();
+        await kbnTestServer.request.get(root, '/route').expect(401);
+      });
+    });
+    describe('when authRequired is `try`', () => {
```

Comment:
I *think* how expired credentials are handling is only an implementation detail of the auth provider, isn't it? 

`AuthResult` is only `Authenticated | AuthNotHandled | AuthRedirected`

and the security auth provider don't seem to really make a distinction either

https://github.com/elastic/kibana/blob/6ecffcc57c9a5b0328257cd25b737140a8471dda/x-pack/plugins/security/server/authentication/authentication_service.ts#L120-L136

Reply:
Sure, but otherwise it's unclear what's the difference between `try` and `optional` modes. It's okay not to add the test if we remove one of them.  https://github.com/elastic/kibana/pull/92784#discussion_r585762695